core/src/main/scala/zio/schema/Schema.scala

```diff
@@ -27,6 +27,410 @@ object Schema {
   sealed case class Tuple[A, B](left: Schema[A], right: Schema[B])       extends Schema[(A, B)]
   final case class EitherSchema[A, B](left: Schema[A], right: Schema[B]) extends Schema[Either[A, B]]
 
+  sealed trait CaseClass[Z] extends Schema[Z] {
```

Comment:
Making sure all the case class types extend this is perfect! This will allow more uniform, type safe handling in user-land code.